DSharpPlus/Clients/DiscordShardedClient.cs

```diff
@@ -230,11 +230,14 @@ public async Task<int> InitializeShardsAsync()
         {
             return this._shards.Count;
         }
-
-        this.GatewayInfo = await this.GetGatewayInfoAsync();
-        int shardCount = this.Configuration.ShardCount == 1 ? this.GatewayInfo.ShardCount : this.Configuration.ShardCount;
+        
         ShardedLoggerFactory loggerFactory = new ShardedLoggerFactory(this.Configuration.LoggerFactory);
         RestClient restClient = new RestClient(this.Configuration, loggerFactory.CreateLogger("Rest"));
+        DiscordApiClient apiClient = new DiscordApiClient(restClient);
```

Comment:
this should be cached for application lifetime so that we don't lose ratelimit information

Reply:
I dont see where we use a api client in the ShardedClient directly and the ratelimit information is saved in the restclient, is it not? And the restclient is used for all clients.

Reply:
the RestClient is created one line above and is absolutely not used for all clients

Reply:
Its passed down to the underlying DiscordClients in the constructor